Download the latest supported chromedriver for the version of Chrome installed by heroku-buildpack-google-chrome

fkztw opened the pull request:
So, we can avoid the chromedriver version mismatch problem in a general situation.

The detail is in: https://github.com/heroku/heroku-buildpack-google-chrome/pull/95

This PR should not be merged until the upstream repo PR been merged.

fkztw commented:
I encountered a problem that there might be no same version of chromedriver for Google Chrome.

As the screenshot of build log below, <https://chromedriver.storage.googleapis.com/85.0.4183.102/chromedriver_linux64.zip> doesn't exist. (The latest version of chromedriver is only `85.0.4183.87` at this time.)

![20200911_17:28:28](https://user-images.githubusercontent.com/1645228/92903239-7d9b9580-f454-11ea-9ad8-35d799f70a85.jpg)

Close this PR at this time until I find a way to deal with this situation.

pvande commented:
From the Chromedriver docs (http://chromedriver.chromium.org/downloads/version-selection):

> Here are the steps to select the version of ChromeDriver to download:
> * First, find out which version of Chrome you are using. Let's say you have Chrome 72.0.3626.81.
> * Take the Chrome version number, remove the last part, and append the result to URL "https://chromedriver.storage.googleapis.com/LATEST_RELEASE_". For example, with Chrome version 72.0.3626.81, you'd get a URL "https://chromedriver.storage.googleapis.com/LATEST_RELEASE_72.0.3626".
> * Use the URL created in the last step to retrieve a small file containing the version of ChromeDriver to use. For example, the above URL will get your a file containing "72.0.3626.69". (The actual number may change in the future, of course.)
> * Use the version number retrieved from the previous step to construct the URL to download ChromeDriver. With version 72.0.3626.69, the URL would be "https://chromedriver.storage.googleapis.com/index.html?path=72.0.3626.69/".
> * After the initial download, it is recommended that you occasionally go through the above process again to see if there are any bug fix releases.

Commit:
Download the latest supported chromedriver for Chrome installed by heroku-buildpack-google-chrome

So, we can avoid the chromedriver version mismatch problem in a general situation.

fkztw commented:
Works like a charm after using the new patch.
Thanks to @pvande.
Downloaded chromedriver 86.0.4240.22 for Google Chrome 86.0.4240.75.
And it works fine on my Heroku app.

![20201012_00:41:35](https://user-images.githubusercontent.com/1645228/95684673-dc0b8d80-0c25-11eb-8598-cc0a1662e0a6.jpg)

How do I modify the `CHANGELOG.md` to pass the "Check changelog" GitHub Action? I know it only checks if there's diff for `CHANGELOG.md`. Is there any contribution guide about the format of the `CHANGELOG.md`? Or the maintainers will modify the `CHANGELOG.md` after reviewing the PR?

And just want to remind again. This PR is staled by https://github.com/heroku/heroku-buildpack-google-chrome/pull/95. Should not be merged before that PR is merged.

edmorley commented:
Hi! Sorry for the delay in replying to this PR. This buildpack isn't an officially supported buildpack so maintenance has been best-effort.

This buildpack has since been replaced by a new combined Chrome+chromedriver buildpack that resolves a number of issues:
https://github.com/heroku/heroku-buildpack-chrome-for-testing

As such this buildpack is being sunset.
